runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/testing/TestDataflowRunner.java

```diff
@@ -174,24 +182,22 @@ DataflowPipelineJob run(Pipeline pipeline, DataflowRunner runner) {
     return runner.apply(transform, input);
   }
 
-  Optional<Boolean> checkForSuccess(DataflowPipelineJob job)
+  Optional<Boolean> checkForSuccess(DataflowPipelineJob job, JobMetrics metrics)
```

Comment:
For streaming case, if job failed or PAssert failed, checkForSuccess() will catch that and return false. Watermark check is not necessary at this time. It only matters when a streaming job doesn't use PAssert. In order to stop waiting as soon as job succeeds, atMaxWatermark will be used.

Will add javadoc to explain return value. But checkForSuccess sounds good to me since it also handles all batch cases.